src/content/docs/en/reference/api-reference.mdx

```diff
@@ -619,6 +619,20 @@ This property is only available when building for SSR (server-side rendering) an
 
 The locale computed from the current URL, using the syntax specified in your `locales` configuration. If the URL does not contain a `/[locale]/` prefix, then the value will default to `i18n.defaultLocale`.
 
+### `Astro.routePattern`
+
+  <Since v="5.0.0" />
+
+The current route pattern being rendered. The value will look like a file path, without the [`srcDir`](/en/reference/configuration-reference/#srcdir) and the `pages` folder. The path will start with a leading slash, and it won't have the file extension.
```

Comment:
```suggestion

The route pattern responsible for generating the current page or route. In file-based routing, this resembles the file path in your project used to create the route. When integrations create routes for your project, `Astro.routePattern` is identical to the value for `injectRoute.pattern`.

The value will start with a leading slash and look similar to the path of a page component relative to your `srcDir/pages/` folder without a file extension.

For example, the file `src/pages/en/blog/[slug].astro` will return `/en/blog/[slug]` for `Astro.routePattern`. Every page on your site generated by that file (e.g. `/en/blog/post-1/`, /en/blog/post-2/`, etc.) shares the same value for `Astro.routePattern`.

You can use this property to understand which route is rendering your component. This allows you to target or analyze similarly-generated page URLs together. For example, you can use it to conditionally render certain information, or collect metrics about which routes are slower.
```

Hi @ematipico ! I was spending some time looking into this, and wondered whether you can tell me how accurate this description above is! Can you correct this where it's wrong? 

I was trying to add more detail to the explanation, so we don't say "routePattern is a route pattern." :smile: and also if there was a cleaner way to describe the pattern part itself without saying "excluding all these things". I also thought it might be helpful to integration authors if we mention what happens with injected routes, so I made a guess there.

So, I might not be correct here! But, can you help me by showing me what's not correct (or useful/important/helpful) with my above understanding! Thank you!

Reply:
Thank you, the suggestion looks good to me. I modified the suggestion to add the following phrase: 
> , `src/pages/index.astro` will return `/`

This is a particular that we should have *somewhere*, because the files that are called `index.*` don't keep the `index` inside the value of `Astro.routePattern`

Reply:
Perfect, thank you! I just updated to give it its own sentence, but this works for me!